torch/csrc/api/src/nn/modules/loss.cpp

```diff
@@ -32,5 +32,30 @@ Tensor HingeEmbeddingLossImpl::forward(
   return F::hinge_embedding_loss(input, target, options);
 }
 
+// ============================================================================
+
+MultiMarginLossImpl::MultiMarginLossImpl(
+    const MultiMarginLossOptions& options_)
+    : options(options_) {
+      reset();
+    }
+
+void MultiMarginLossImpl::reset() {
+  register_buffer("weight", options.weight().value());
```

Comment:
It'd be awesome if we also add the following checks (expressed in Python), to match the Python version better:
```python
        if p != 1 and p != 2:
            raise ValueError("only p == 1 and p == 2 supported")
        assert weight is None or weight.dim() == 1
```

Reply:
Is there a reason to treat these two errors with `raise` and `assert`? In the next comment, both errors are treated using `raise`. In any case, looking at how other errors are handled, I used the `TORCH_CHECK` macro, I hope that was the right way!

Reply:
Yes using `TORCH_CHECK` is the right way - `raise` and `assert` are both Python syntax and we should use the C++ equivalent. Thanks a lot for your help!